sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java

```diff
@@ -123,26 +145,75 @@ public void close() {
 
     private Mono<Void> sendClientTelemetry() {
         return Mono.delay(Duration.ofSeconds(clientTelemetrySchedulingSec), CosmosSchedulers.COSMOS_PARALLEL)
-                   .flatMap(t -> {
-                       if (this.isClosed) {
-                           logger.warn("client already closed");
-                           return Mono.empty();
-                       }
-
-                       if(!Configs.isClientTelemetryEnabled(this.isClientTelemetryEnabled)) {
-                           logger.trace("client telemetry not enabled");
-                           return Mono.empty();
-                       }
-
-                       readHistogram();
-                       try {
-                           logger.info("ClientTelemetry {}", OBJECT_MAPPER.writeValueAsString(this.clientTelemetryInfo));
-                       } catch (JsonProcessingException e) {
-                           logger.error("Error which parsing client telemetry into json. ", e);
-                       }
-                       clearDataForNextRun();
-                       return this.sendClientTelemetry();
-                   }).onErrorResume(ex -> {
+            .flatMap(t -> {
+                if (this.isClosed) {
+                    logger.warn("client already closed");
+                    return Mono.empty();
+                }
+
+                if (!Configs.isClientTelemetryEnabled(this.isClientTelemetryEnabled)) {
+                    logger.trace("client telemetry not enabled");
+                    return Mono.empty();
+                }
+                readHistogram();
+                try {
+                    String endpoint = Configs.getClientTelemetryEndpoint();
```

Comment:
I would cache `Configs.getClientTelemetryEndpoint()` value in `ClientTelemetry`. to not pay the additional overhead of `System.getProperty(CLIENT_TELEMETRY_ENDPOINT)` each time. 

do we expect the value of `getClientTelemetryEndpoint` change at runtime? is that why this is not set as field in ClientTelemetry, or any other reason?

Reply:
That was the intention to keep clientTelemetryEnabled and clientTelemetryEnpoint changeable on the runtime. This is one call in 10 min , so not an issue. But your point is more applicable to TracerProvider and CosmosPageFlux where we are calling Configs.isClientTelemetryEnabled() on every response. However during the perf testing we did not overserve any perf hit. We should talk offline. Also I noticed in gateway call (http call ) we are calling config methods on every call including query plan

Reply:
Resolving as per offline conversation , this need to be changed all across sdk at once